x/tokenfactory/keeper/before_send_test.go

```diff
@@ -0,0 +1,71 @@
+package keeper_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/neutron-org/neutron/v3/x/tokenfactory/types"
+)
+
+func (suite *KeeperTestSuite) TestTrackBeforeSendWasm() {
+	for _, tc := range []struct {
+		name     string
+		wasmFile string
+	}{
+		{
+			name:     "Test bank hook tracking contract ",
+			wasmFile: "./testdata/bank_hook_test.wasm",
```

Comment:
it would be helpful if you put link to the sources here

Reply:
Done